src/vendor/azure-uamqp-c/src/message.c

```diff
@@ -1447,3 +1465,71 @@ int message_get_message_format(MESSAGE_HANDLE message, uint32_t *message_format)
 
     return result;
 }
+
+int message_set_delivery_tag(MESSAGE_HANDLE message, AMQP_VALUE delivery_tag_value)
+{
+    int result;
+
+    if ((message == NULL) || (delivery_tag_value == NULL))
+    {
+        LogError("Bad arguments: message = %p, delivery_tag = %p",
+            message, delivery_tag_value);
+        result = __FAILURE__;
+    }
+    else
+    {
+        AMQP_VALUE new_delivery_tag = amqpvalue_clone(delivery_tag_value);
+        if (new_delivery_tag == NULL)
+        {
+            LogError("Cannot clone delivery tag");
+            result = __FAILURE__;
+        }
+        else
+        {
+
```

Comment:
should we add this here?
```c
if (message->delivery_tag != NULL)
{
    amqpvalue_destroy(message->delivery_tag);
}
```

Reply:
I can't tell where you're suggesting I put it - I think the github diff is messed up....

Reply:
Ah I get it - again I don't think it's necessary as this value should never be set twice, but will add to be extra safe